internal/stacks/stackruntime/internal/stackeval/component_config.go

```diff
@@ -606,6 +540,10 @@ func (w *sourceBundleModuleWalker) finalSourceForModule(tfSourceAddr addrs.Modul
 	}
 
 	switch sourceAddr := sourceAddr.(type) {
+	case sourceaddrs.LocalSource:
+		// Local sources must be combined with the root module source to give
+		// an absolute address.
+		return sourceaddrs.ResolveRelativeFinalSource(w.rootModuleSource, sourceAddr)
```

Comment:
Is it correct to _always_ use the "root" module source? I am not sure I fully followed all the data flow here and so I might be misunderstanding, but I worry this wouldn't be correct if, for example:

1. root module calls a remote source address specifying a different module package than the root
2. that other module in turn calls something else in its own package using a relative path

This seems like it might end up resolving the relative path against the root module's source package, rather than against the second-level module's source package. :thinking:

Reply:
Excellent catch, thank you! I think I've addressed this in 7d956bbc11; please let me know if that's unclear or if it's not solving the whole problem.

(I am a bit confused about why sourceaddrs aren't _always_ fully qualified but that's not something we need to address here unless it's pertinent.)